lib/esp-scanning-lib.robot

```diff
@@ -0,0 +1,173 @@
+*** Settings ***
+Documentation       Collection of keywords related to UEFI Secure Boot
+
+Resource            ../keywords.robot
+
+
+*** Variables ***
+${SHIMX64_URL}=     https://cloud.3mdeb.com/index.php/s/TipPHgsCj6MCZHi/download/shimx64.efi
+${GRUBX64_URL}=     https://cloud.3mdeb.com/index.php/s/e2JjYANjMsgyXtS/download/grubx64.efi
+${TINYCORE_URL}=    https://distro.ibiblio.org/tinycorelinux/14.x/x86/release/CorePlus-current.iso
+${DTS_URL}=         https://dl.3mdeb.com/open-source-firmware/DTS/v1.2.8/dts-base-image-v1.2.8.iso
+
+
+*** Keywords ***
+Prepare EFI Partition With System Files
+    [Documentation]    Prepares a bootable EFI partition with OS files
+    ...    by mounting a hard drive in Qemu, or creating files on the DUT
+    ...    via PiKVM
+
+    IF    "${MANUFACTURER}" == "QEMU"
+        Add HDD To Qemu    img_name=image.img
+    ELSE
+        IF    "${DUT_CONNECTION_METHOD}" == "pikvm"
+            Boot System Or From Connected Disk    ubuntu
+            Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+            Login To Linux
+            Switch To Root User
+            Remove All Supported Systems From Efi
+            Populate /EFI With System Folders
+            Set Global Variable    ${DUT_CONNECTION_METHOD}    pikvm
+        ELSE
+            Skip    unsupported
+        END
+    END
+
+Clear Out EFI Partition
+    [Documentation]    Prepares a bootable EFI partition with OS files
+    ...    by unmounting the hard drive in Qemu, or erasing files on the DUT
+    ...    via PiKVM
+
+    IF    "${MANUFACTURER}" == "QEMU"
+        Remove Drive From Qemu
+    ELSE
+        IF    "${DUT_CONNECTION_METHOD}" == "pikvm"
+            Boot System Or From Connected Disk    ubuntu
+            Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+            Login To Linux
+            Switch To Root User
+            Remove All Supported Systems From Efi
+            Set Global Variable    ${DUT_CONNECTION_METHOD}    pikvm
```

Comment:
We have a keyword: `Restore Initial DUT Connection Method`

Reply:
Resolved via `Execute Reboot Command` in [commit a90d5e0](https://github.com/Dasharo/open-source-firmware-validation/commit/a90d5e01100174c8b883b01ae914d4693f595747#diff-c8f7a5cdc7c64ba736049ef48cf813ae918ab342659418e21d720981c53f903aR38)